app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -450,16 +450,17 @@ export const getSimulatedCode = async (ethereumClientService: EthereumClientServ
 		input: input,
 		accessList: []
 	} as const
-
-	const simulatedTransactions = await simulateTransactionsOnTopOfSimulationState(ethereumClientService, requestAbortController, simulationState, [getCodeTransaction], { [addressString(GET_CODE_CONTRACT)]: { code: getCodeByteCode() } })
-	const lastResult = simulatedTransactions.at(-1)
-	if (lastResult === undefined) throw new Error('last result did not exist in multicall')
-	if (lastResult.status === 'failure') return { statusCode: 'failure' } as const
-	const parsed = atInterface.decodeFunctionResult('at', lastResult.returnData)
-	return {
-		statusCode: lastResult.status,
-		getCodeReturn: EthereumData.parse(parsed.toString())
-	} as const
+	try {
+		const simulatedTransactions = await simulateTransactionsOnTopOfSimulationState(ethereumClientService, requestAbortController, simulationState, [getCodeTransaction], { [addressString(GET_CODE_CONTRACT)]: { code: getCodeByteCode() } })
+		const lastResult = simulatedTransactions.at(-1)
+		if (lastResult === undefined) throw new Error('last result did not exist in multicall')
+		if (lastResult.status === 'failure') return { statusCode: 'failure' } as const
+		const parsed = atInterface.decodeFunctionResult('at', lastResult.returnData)
+		return { statusCode: lastResult.status, getCodeReturn: EthereumData.parse(parsed.toString()) } as const
+	} catch(error: unknown) {
+		if (error instanceof JsonRpcResponseError) return { statusCode: 'failure' } as const
+		throw error
+	}
```

Comment:
As we have discussed previously, it feels like we should not be using try/catch to handle code flow.  Erorrs from a remote server should be handled as part of normal code paths, not as part of exceptional code paths.  Is there no way to detect an error response without throwing in our current codebase?